client/src/hooks/useWindowSize.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+type WindowSize = {
+    innerWidth: number;
+    innerHeight: number;
+}
+function getWindowSize():WindowSize {
+	const { innerWidth, innerHeight } = window;
+	return {
+		innerWidth, innerHeight
+	};
+}
+function useWindowSize():WindowSize {
+	const [windowSize, setWindowSize] = React.useState(getWindowSize());
+
+	React.useEffect(() => {
```

Comment:
async ?

Reply:
good Idea.

Reply:
nvm bad idea, hooks should be syncronous. if not it will break the whole ui.

Reply:
> async ?

I also tried but bc its async it will break the whole thing bc it will get either null or a promise.